octopus/src/test/scala/octopus/ValidationSpec.scala

```diff
@@ -352,6 +352,24 @@ class ValidationSpec
         )
       }
     }
+
+    "given comap" should {
+
+      "validate lifted type" in {
+        case class Lifted(value: String)
+        val unlift: Lifted => String = _.value
+
+        val validator = Validator[String].rule(!_.isEmpty, "empty")
```

Comment:
You can simplify to single expression:

`implicit val liftedValidator = Validator[String].rule(...).comap(unlift)`